runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -248,6 +248,13 @@ end
 ---@param line integer
 ---@param is_spell_nav boolean
 local function on_line_impl(self, buf, line, is_spell_nav)
+  -- Track the maximum pattern index encountered in each tree. For subsequent
+  -- trees, the subpriority passed to nvim_buf_set_extmark is offset by the
+  -- largest pattern index from the prior tree. This ensures that extmarks
+  -- from subsequent trees always appear "on top of" extmarks from previous
+  -- trees (e.g. injections should always appear over base highlights).
```

Comment:
This specifically isn't working for me. Which part of the code aims to ensure injection highlights takes priority?

Reply:
Ok, I've debugged the issue. The problem is `on_line_impl` can (and often does) highlight more than 1 line.

My example:

```bash
echo "
module dut (input clk, reset_n);
endmodule
" > dut.sv
```

Here the contents of `echo` are an injection. However the base tree will highlight the string argument to `echo` in the call to `on_line_imp(line=0)`.

When `on_line_impl(line=1)` is called, there is no highlight state in the base tree and so when the injection is highlighted, `pattern_offset=0`.

This causes the highlight from the base tree to have a higher subpriority than the injection.

So basically I don't think this method is going to work. What might work is offsetting the priority based on how many parents the langtree has.

Reply:
https://github.com/neovim/neovim/pull/27844 should fix it.